### PR TITLE
Add select/deselect all style categories

### PR DESCRIPTION
**Pull request by ghtmtt**
Adds two buttons to select and deselect all style categories when saving or loading a style:

image

image

### CODE REVIEW

**Comment by roya0045** [comment]
Adding the option to select multiple rows and toggle them with the spacebar would also speed up the process when importing and saving styles.

**Comment by nyalldawson** [comment]
Nice! Can you reorder the buttons to "Select All" -> "Deselect All"? 

We are inconsistent with this order, but there's slightly more places were the Select is before the Deselect. I'll unify these in a different PR.

File: src/gui/qgsmaplayersavestyledialog.h [hunk 1]
```diff
@@ -104,6 +104,8 @@ class GUI_EXPORT QgsMapLayerSaveStyleDialog : public QDialog, private Ui::QgsMap
 
   public slots:
     void accept() override;
+    void showAll();
+    void hideAll();
```
Comment (nyalldawson) [inline src/gui/qgsmaplayersavestyledialog.h:108]:
```suggestion
    void selectAll();
    void deselectAll();
```

These should also be private slots, not public

File: src/gui/qgsmaplayersavestyledialog.cpp [hunk 2]
```diff
@@ -76,12 +76,34 @@ QgsMapLayerSaveStyleDialog::QgsMapLayerSaveStyleDialog( QgsMapLayer *layer, QWid
   mModel->setCategories( lastStyleCategories );
   mStyleCategoriesListView->setModel( mModel );
 
+  // select and deselect all categories
+  connect( mShowAllButton, &QPushButton::clicked, this, &QgsMapLayerSaveStyleDialog::showAll );
+  connect( mHideAllButton, &QPushButton::clicked, this, &QgsMapLayerSaveStyleDialog::hideAll );
```
Comment (nyalldawson) [inline src/gui/qgsmaplayersavestyledialog.cpp:81]:
```suggestion
  connect( mSelectAllButton, &QPushButton::clicked, this, &QgsMapLayerSaveStyleDialog::showAll );
  connect( mDeselectAllButton, &QPushButton::clicked, this, &QgsMapLayerSaveStyleDialog::hideAll );
```

**Comment by ghtmtt** [comment]
@nyalldawson thanks for the review! I should have done what you suggested with the buttons reordered and function names adapted in the code.

**Comment by ghtmtt** [comment]
@roya0045 I've added also an `Invert Selection` button. @nyalldawson I've moved to private slots

**Comment by ghtmtt** [comment]
Just for reference of the last `Invert Selection` button:

image

image

**Comment by DelazJ** [comment]
@ghtmtt Could you instead add the new images in your top message? This is the one that will be picked for docs and changelog, so better have the up to date screenshots there...

**Comment by ghtmtt** [comment]
@DelazJ done!

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**